spotbugs-tests/src/test/java/edu/umd/cs/findbugs/ba/ValueRangeDataflowTest.java

```diff
@@ -0,0 +1,253 @@
+
+public class ValueRangeDataflowTest extends AbstractIntegrationTest {
+
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private final PrintStream originalOut = System.out;
+
+    @Before
+    public void setUpStreams() {
+        System.setOut(new PrintStream(outContent));
+    }
+
+    @After
+    public void restoreStreams() {
+        System.setOut(originalOut);
+    }
+
+    @Test
+    public void test1() {
+        System.setProperty("dataflow.classname", "edu.umd.cs.findbugs.ba.ValueRangeDataflow");
+        System.setProperty("dataflow.printcfg", "true");
+        performAnalysis("ValueRangeDataflow.class");
+        String[] outputLines = outContent.toString().split(System.lineSeparator());
+        checkOutput(outputLines);
+    }
+
+    private void checkOutput(String[] outputLines) {
+        for (int i = 0; i < outputLines.length; ++i) {
+            if ("Method: ValueRangeDataflow.eq(int)".equals(outputLines[i])) {
+                checkEQ(outputLines, i + 4);
+            } else if ("Method: ValueRangeDataflow.ne(int)".equals(outputLines[i])) {
+                checkNE(outputLines, i + 4);
+            } else if ("Method: ValueRangeDataflow.lt(int)".equals(outputLines[i])) {
+                checkLT(outputLines, i + 4);
+            } else if ("Method: ValueRangeDataflow.le(int)".equals(outputLines[i])) {
+                checkLE(outputLines, i + 4);
+            } else if ("Method: ValueRangeDataflow.gt(int)".equals(outputLines[i])) {
+                checkGT(outputLines, i + 4);
+            } else if ("Method: ValueRangeDataflow.ge(int)".equals(outputLines[i])) {
+                checkGE(outputLines, i + 4);
+
+            } else if ("Method: ValueRangeDataflow.eqBoolean(boolean)".equals(outputLines[i])) {
+                checkEQBoolean(outputLines, i + 4);
+            } else if ("Method: ValueRangeDataflow.eqByte(byte)".equals(outputLines[i])) {
+                checkEQByte(outputLines, i + 4);
+            } else if ("Method: ValueRangeDataflow.eqShort(short)".equals(outputLines[i])) {
+                checkEQShort(outputLines, i + 4);
+            } else if ("Method: ValueRangeDataflow.eqLong(long)".equals(outputLines[i])) {
+                checkEQLong(outputLines, i + 4);
+            } else if ("Method: ValueRangeDataflow.eqChar(char)".equals(outputLines[i])) {
+                checkEQChar(outputLines, i + 4);
+
+            } else if ("Method: ValueRangeDataflow.fullRange(int)".equals(outputLines[i])) {
+                checkFullRange(outputLines, i + 4);
+            } else if ("Method: ValueRangeDataflow.partialRange(int)".equals(outputLines[i])) {
+                checkPartialRange(outputLines, i + 4);
+
+            } else if ("Method: ValueRangeDataflow.otherRange(int, short)".equals(outputLines[i])) {
+                checkOtherRange(outputLines, i + 4);
+            }
+        }
+    }
+
+    private void checkEQ(String[] outputLines, int n) {
```

Comment:
What is `n`? A more descriptive name would be good. Especially with the cryptic method name.